src/actions/guides/handling_files/file_uploads.cr

```diff
@@ -0,0 +1,104 @@
+class Guides::HandlingFiles::FileUploads < GuideAction
+  guide_route "/handling-files/file-uploads"
+
+  def self.title
+    "Uploading Files"
+  end
+
+  def markdown : String
+    <<-MD
+    ## Setup
+
+    For handling file uploads, we will use [Shrine.cr](https://github.com/jetrockets/shrine.cr).
+
+    Start by adding the `shrine` shard to your `shard.yml` and run `shards install`.
+
+    ```yaml
+    dependencies:
+      shrine:
+        github: jetrockets/shrine.cr
+    ```
+
+    Next you will require the shard in `src/shards.cr`
+
+    ```crystal
+    # src/shards.cr
+
+    # Require your shards here
+    require "avram"
+    require "lucky"
+    require "carbon"
+    require "authentic"
+    require "shrine"
+    ```
+
+    Lastly, we can create a config file for configuring where our uploaded files will be stored.
+    Create a new file in `confg/shrine.cr`
+
+    ```crystal
+    # config/shrine.cr
+    Shrine.configure do |config|
+      config.storages["cache"] = Storage::FileSystem.new("uploads", prefix: "cache")
+      config.storages["store"] = Storage::FileSystem.new("uploads")
+    end
+    ```
+
+    ## SaveOperation setup
+
+    The `file_attribute` is used in your save operation to specify the name of the param attribute that will contain the file.
+
+    ```crystal
+    class SaveUser < User::SaveOperation
+      permit_columns name
+      file_attribute :profile_picture
+
+      before_save do
+        profile_picture.value.try { |pic| upload_pic(pic) }
+      end
+
+      private def upload_pic(pic)
+        result = Shrine.upload(File.read(pic.tempfile.path), "store", metadata: { "filename" => pic.filename })
+        profile_picture_path.value = result.url
```

Comment:
```suggestion
        profile_picture_path.value = result.id
```

Storing the `id` instead of the `url` allows you to use `Shrine` later (like if you want to delete the uploaded file).  Shrine's docs actually have you turn the whole `result` into json and store that but 🤷